### PR TITLE
clientv3/integration: fix TestSetEndpointAndPut dial tests

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/8859.

### CODE REVIEW

File: clientv3/integration/dial_test.go [hunk 1]
```diff
@@ -191,15 +191,19 @@ func TestDialForeignEndpoint(t *testing.T) {
...
 	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 2})
 	defer clus.Terminate(t)
 
 	clus.Client(1).SetEndpoints(clus.Members[0].GRPCAddr())
 	_, err := clus.Client(1).Put(context.TODO(), "foo", "bar")
-	if err != nil && !strings.Contains(err.Error(), "closing") {
+	// grpc/transport.ErrConnClosing or grpc/transport.ErrConnDrain on balancer endpoint switch
+	if err != nil &&
+		!strings.Contains(err.Error(), "transport is closing") &&
+		!strings.Contains(err.Error(), "the server stops accepting new RPCs") {
```
Comment (xiang90) [inline clientv3/integration/dial_test.go]: why we might receive a server side error triggered by a endpoint switch?

Comment (gyuho) [reply clientv3/integration/dial_test.go]: Balancer switch might not have completed, so errors return while gRPC drains current connection and creates a new sub connection.

Comment (xiang90) [reply clientv3/integration/dial_test.go]: transport is closing makes sense. i was asking why we can see `the server stops...`. gRPC client draining connection has nothing to do with the server.

Comment (gyuho) [reply clientv3/integration/dial_test.go]: > gRPC client draining connection has nothing to do with the server.

This error message is more about gRPC http/2 client telling the current connection has been drained.

1. `tearDown` on current `*addrConn(errConnDrain)`
2. `ac.transport.GracefulClose()` that prevents new streams from being created, setting the connection state to drain

So, it is the `http2Client` that returns `ErrStreamDrain` error, not from the server.

**Comment by gyuho** [comment]
This needs rework after https://github.com/grpc/grpc-go/pull/1844 is merged.

**Pull request closed without merging**